### DISCUSSION TITLE
RXVERS assembler instruction

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#344] [Q&A]
I currently work on a new assembler instruction to return basic architecture information. I haven't found a way to get the underlying operation system. For 32/64 bit there is a precompiler instruction `#ifdef __32BIT__` which can be used. for the OSes the flags _`_WINDOWS__, LINUX__, __APPLE__` are not passed. Any idea how this could be achieved?

this is what we have so far: 
```
assembler RXVERS version
say version 
```
`UNKNOWN 64BIT cREXX F0044`

**Comment by adesutherland** [comment]
OK - I have tweaked what you have done and it is just been pushed into develop
